File: csharp/Behavioral/22-Visitor/Element/ElectricalJob.cs
namespace Visitor;

public class ElectricalJob : IJobElement
{
    public string JobId { get; } = Guid.NewGuid().ToString("N")[..8];

    public string Title { get; }
    public decimal Cost { get; }
    public decimal CircuitLoadAmps { get; }
    public bool RequiresCertificate { get; }

    public ElectricalJob(
        string title, decimal cost,
        decimal circuitLoadAmps, bool requiresCertificate)
    {
        Title               = title;
        Cost                = cost;
        CircuitLoadAmps     = circuitLoadAmps;
        RequiresCertificate = requiresCertificate;
    }

    public void Accept(IJobVisitor visitor) => visitor.Visit(this);
}

File: typescript/Behavioral/22-Visitor/Element/ElectricalJob.ts
import type { IJobElement } from "./IJobElement";
import type { IJobVisitor } from "../Visitor/IJobVisitor";

export class ElectricalJob implements IJobElement {
  readonly jobId: string = Math.random().toString(36).slice(2, 10);

  constructor(
    readonly title: string,
    readonly cost: number,
    readonly circuitLoadAmps: number,
    readonly requiresCertificate: boolean,
  ) {}

  accept(visitor: IJobVisitor): void {
    visitor.visitElectrical(this);
  }
}
